pkg/apis/druid/v1alpha1/druid_types.go

```diff
@@ -141,6 +141,9 @@ type DruidNodeSpec struct {
 	// Required: Port used by Druid Process
 	DruidPort int32 `json:"druid.port"`
 
+	// Required: Define kind for node. Deployment or Statefulsets are supported
+	Kind string `json:"kind"`
```

Comment:
can we make it optional and default to `StatefulSet` to keep things backward compatible and also to not necessarily having to specify this.

Reply:
sure, makes sense to have backward compatible. Ill do the changes :)